src/include/sof/audio/buffer.h

```diff
@@ -83,9 +84,21 @@ extern struct tr_ctx buffer_tr;
 #define BUFF_PARAMS_RATE	BIT(2)
 #define BUFF_PARAMS_CHANNELS	BIT(3)
 
-/* audio component buffer - connects 2 audio components together in pipeline */
+/*
+ * audio component buffer - connects 2 audio components together in pipeline.
+ *
+ * The buffer is a hot structure that must be shared on certain cache
+ * incoherent architectures.
```

Comment:
Shouldn't there be some property that actually checks if the buffer will be used between two cores? What happens in the typical case where the entire pipeline runs on one core? then all this cache coherency is not really needed except for DMA buffers, is it?

Put differently, is the multi-core support going to impact or possible cripple performance when chaining processing on a single core?

Reply:
minimal impact for non shared objects - we still check the ```c->shared``` flag, so lock and cache ops are only performed on objects used by multiple cores.
```
__must_check static inline struct coherent *coherent_acquire(struct coherent *c,
							     const size_t size)
{
	/* assert if someone passes a cache/local address in here. */
	ADDR_IS_COHERENT(c);

	/* this flavour should not be used in IRQ context */
	CHECK_COHERENT_IRQ(c);

	/* access the shared coherent object */
	if (c->shared) {
		CHECK_COHERENT_CORE(c);

		spin_lock(&c->lock);

		/* invalidate local copy */
		dcache_invalidate_region(uncache_to_cache(c), size);
	}

	/* client can now use cached object safely */
	return uncache_to_cache(c);
}
```